modules/pytket-qiskit/pytket/extensions/qiskit/backends/aer.py

```diff
@@ -156,19 +156,16 @@ def backend_info(self) -> BackendInfo:
     def process_circuits(
         self,
         circuits: Sequence[Circuit],
-        n_shots: Optional[Union[int, Sequence[int]]] = None,
+        n_shots: Union[None, int, Sequence[Optional[int]]] = None,
         valid_check: bool = True,
         **kwargs: KwargTypes,
     ) -> List[ResultHandle]:
         circuits = list(circuits)
-        n_shots_list: List[Optional[int]] = []
-        if hasattr(n_shots, "__iter__"):
-            n_shots_list = cast(List[Optional[int]], n_shots)
-            if len(n_shots_list) != len(circuits):
-                raise ValueError("The length of n_shots and circuits must match")
-        else:
-            # convert n_shots to a list
-            n_shots_list = [cast(int, n_shots)] * len(circuits)
+        n_shots_list = Backend._get_n_shots_as_list(
```

Comment:
Why are you doing a cast only sometimes?

Reply:
Thank you for explaining that. If you add the removing of the cast to the task, I think it is fine to merge it in now.